src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/AnyRef.cs

```diff
@@ -2,26 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-using System.Threading;
+using System.Diagnostics.CodeAnalysis;
 using Microsoft.Win32.SafeHandles;
 
 namespace System.Runtime.InteropServices.JavaScript
 {
-    public abstract class AnyRef : SafeHandleMinusOneIsInvalid
+    public partial class JSObject : SafeHandleMinusOneIsInvalid
     {
         private GCHandle? InFlight;
         private int InFlightCounter;
-        private GCHandle AnyRefHandle;
         public int JSHandle => (int)handle;
+        public bool IsDisposed { get; private set; }
 
-        internal AnyRef(IntPtr jsHandle, bool ownsHandle) : base(ownsHandle)
+        public JSObject() : base(true)
+        {
+            InFlight = null;
+            InFlightCounter = 0;
+
+            var jsHandle = Runtime.CreateCsOwnedObject(this, nameof(Object));
```

Comment:
What's the actual typename we intend here? I don't think nameof is appropriate

Reply:
It's refactored of original [`Interop.Runtime.New<object>()`](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.cs#L28) which does [` New(typeof(T).Name, parms, out int exception)`](https://github.com/dotnet/runtime/blob/e49b9bd7ae706bbcbefc767d6572232acbb20940/src/libraries/Common/src/Interop/Browser/Interop.Runtime.cs#L80), which does [`var coreObj = globalThis[js_name]`](https://github.com/dotnet/runtime/blob/e49b9bd7ae706bbcbefc767d6572232acbb20940/src/mono/wasm/runtime/binding_support.js#L1957) which is effectively `new globalThis.Object()`

Reply:
OK, makes sense. I think it would be ideal if we stopped doing that, and explicitly specified the name of the JS type as a string. Using nameof will cause any C# refactorings to change the JS name, which is never what we want